modules/dnn/src/dnn_common.hpp

```diff
@@ -154,6 +154,21 @@ static inline std::string toString(const Mat& blob, const std::string& name = st
     return ss.str();
 }
 
+// Scalefactor is a common parameter used for data scaling. In OpenCV, we often use Scalar to represent it.
+// Because 0 is meaningless in scalefactor.
+// If the scalefactor is (x, 0, 0, 0), we convert it to (x, x, x, x). The following func will do this hack.
+Scalar hackScalar(const Scalar& _scale)
```

Comment:
Should be `static` / `static inline` or moved to .cpp

Reply:
can we use some better name? say, broadcastRealScalar?

Reply:
got it